### DISCUSSION TITLE
Fundamentos do C#

### DISCUSSION THREAD

**Discussion by andrebaltieri** [balta-io/2801#2] [Perguntas e Respostas] [answered]
Este fórum é destinado a perguntas e respostas sobre o curso Fundamentos do C#.

**Comment by Rafael354-dev** [comment, marked as answer]
Olá! Bom dia!
Estou na aula de fundamentos de C# e, não estou conseguindo acessar o caminho para colocar no arquivo digitado no C:

**Reply by andrebaltieri** [reply]
Boa tarde @Rafael354-dev como vai?

Para te ajudar preciso de bem mais detalhes... qual caminho? Qual aula? QUal erro você recebe?

**Reply by Rafael354-dev** [reply]
Descobri qual foi o erro, não estava colocando o texto.txt no final.
na aula de editar texto.

**Comment by DeividStenico** [comment]
Bom dia, estou na aula de Fundamentos, mais especificamente na aula de Debug (6~7 minutos). Quando clico no ícone do "Run and Debug", acho que devido meu VSCode ser mais atualizado que o da aula e o .net já estar na versão 5, não sei o motivo, mas o Debug não executa... Me aparece uma mensagem para criar um "launch.json". Se clico pra criar, abre o arquivo com essa linha de comando:
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": []
}

Ou se clico em Run and Debug logo acima, abre essa janela:
![debug](https://user-images.githubusercontent.com/90517291/132952095-dcc25347-bacf-4a9c-afd4-b167c7f8f90a.jpg)

E mais adiante na mesma aula, tem meio que essa questão se não aparecer o ícone na parte superior e tal... Mas se eu clicar em .NET Core, abre o arquivo "launch.json", mas com um erro logo abaixo, segue:
![debug2](https://user-images.githubusercontent.com/90517291/132952393-11623e6b-69db-427f-abb4-9c35b2b028cc.jpg)


Poderia me ajudar nesse início? Muito obrigado.

**Reply by andrebaltieri** [reply]
Boa tarde @DeividStenico, conforme mostrado no erro, não tem como debugar antes do OmniSharp (Debugador) ser inicializado.
Aguarde uns minutos e tente novamente... Se for o caso, verifica se o OmniSharp está em execução ai... o ícone dele é um fogo (🔥) que fica no canto inferior esquerdo do VSCode.

**Reply by denisbao** [reply]
@DeividStenico, eu acabei encontrando o mesmo problema.

No meu caso, parece ser apenas o VSCode tentando se entender com as extensões.
Consegui resolver o problema criando o projeto novo projeto, após já ter instalado a extensão "C#" no VSCode.

**Reply by jvgsfilho** [reply]
@DeividStenico Passei pelo mesmo problema e consegui resolver.

Quando voce criar o json, no canto inferior esquerdo tem um botão chamado "Add Configuration"
![image](https://user-images.githubusercontent.com/31940719/134263563-2e0c304f-d170-4931-9442-68b6e14abfc7.png)

Escolhe a 3 opção
![image](https://user-images.githubusercontent.com/31940719/134263606-b9622bb0-2370-448b-8200-e952d44bb205.png)

Substitui os itens entre chaves pelo framework e o nome do projeto (nome da pasta)
No meu caso ficou assim
"program": "${workspaceFolder}/bin/Debug/net5.0/Teste2.dll",
![image](https://user-images.githubusercontent.com/31940719/134263729-830df7d4-f3f2-4891-98d8-97e18317fb6a.png)

Após isso, salva o json (ctrl + s) e tenta debugar que vai funcionar .

**Reply by jvgsfilho** [reply]
achei uma maneira mais fácil.
ao abrir o programa pela primeira vez, acessa a linha de comando (Ctrl + SHIFT + P) e seleciona a opção abaixo:
![image](https://user-images.githubusercontent.com/31940719/135004933-87782c68-8ebc-4ff6-8621-7786adda6a74.png)

Após selecionar seu projeto, vai em Debug e peça para criar o JSON, dessa vez ele vai criar automático sem precisar ter que colocar o nome manualmente.

**Comment by eduardodsr** [comment]
Boa noite.

Tudo bem.
No curso (**Fundamentos do C#** -> **Mão na Massa: Editor de Texto** -> Aula - Salvando um arquivo).
Na parte prática de Salvar o arquivo no Editor de Texto, eu não consigo salvar o arquivo gerado na aula. 
Utilizo o sistema operacional MacOS. Até criei uma pasta com permissão, mas apresenta  informando que não tenho permissão. Procurei na Internet mais não encontrei uma solução. Por favor tem como ajudar?

Seguinte a mensagem de erro: "---> System.IO.IOException: Permission denied:

Atenciosamente,
Eduardo Rodrigues

**Reply by andrebaltieri** [reply]
Tenta `sudo dotnet run`

**Comment by leoothomazz** [comment]
Olá Professor, tenho uma dúvida conceitual sobre a aula de Timezone...

Você citou na aula, que podemos utilizar o Timezone info de uma localidade, tal qual na aula você apresenta o da Austrália...
No entanto, foi citado que os timezones disponíveis variam de acordo com o sistema operacional.

Imaginando que uma aplicação minha de produção esteja hospedada em um servidor qualquer, ou até mesmo replicada em um servidor que seja Windows, e outro que seja Linux...como eu faria pra saber que estou utilizando o Timezone de maneira correta em ambos os servidores para que minhas conversões ocorram de maneira correta?

Além disso, não corro o risco do timezone possuir um nome diferente em cada tipo de servidor? - Como eu poderia validar para fazer essas conversões corretamente, independente do sistema operacional do servidor em que a aplicação esteja hospedada?

**Comment by andrebaltieri** [comment]
Bom dia @leoothomazz ótima dúvida!

Vamos supor que seu sistema rode em Windows (EN-US) aqui no BR e Linux (EN-US) na Europa... a hora que ele vai considerar como base é sempre a do SO, logo estes sistemas teriam o mesmo horário.

Agora, uma boa prática para garantir isto é sempre salvar as datas no formato UTC no banco e na hora de ler, localizar elas para a cultura do usuário.

Sobre os nomes dos Timezones, corre sim, inclusive são diferentes no Windows e Linux, é sempre importante listar eles antes para saber se está utilizando o correto caso necessário, ou então usar o esquema de UTC que mencionei acima.

**Comment by Leonardocorreabrasil** [comment]
Bom dia Professor, tudo bem?

comecei a estudar através do curso de fundamento de c#, estou fazendo uns exercícios para praticar, tem um em especifico 
que o valor de ponto de flutuante que 
não é o esperado, consegues me dar um help? como mostra a foto abaixo:
![erro-numerosFlutuantes](https://user-images.githubusercontent.com/54727753/148049150-262d17da-e7bd-4ecc-9fcc-e59e6096c31b.PNG)

**Reply by andrebaltieri** [reply]
@Leonardocorreabrasil você chegou a debugar e ver qual valor capturado no `Console.Read`?

**Comment by Rafael354-dev** [comment]
Consegui instalar era erro de digitação, não percebi.

Porém agora quando digito no terminal ele da outro erro.

***@***.***

Enviado do Email<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: Andre ***@***.***>
Enviado:terça-feira, 4 de janeiro de 2022 09:41
Para: ***@***.***>
Cc:Rafael Gianini ***@***.***>; ***@***.***>
Assunto: Re: [balta-io/2801] Fundamentos do C# (Discussion #2)


@Leonardocorreabrasil<https://github.com/Leonardocorreabrasil> você chegou a debugar e ver qual valor capturado no Console.Read?

—
Reply to this email directly, view it on GitHub<https://github.com/balta-io/2801/discussions/2#discussioncomment-1905628>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ATYS5UNBEQLWLEUUZ4YIP33UULTGTANCNFSM47UCVMGQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by andrebaltieri** [reply]
@Rafael354-dev e qual seria o erro?

**Comment by Rafael354-dev** [comment]
Então balta, na verdade eu digitei nodemon errado, porem agora q instalou e vou colocar o nodemon  pra start da outro erro, no curso vc colocou pra salvar npm intall nodemon  –save-dev. Se eu usar outro caminho igual  estou usando no anexo, tem problema?

Aguardo seu retorno.


Enviado do Email<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: Andre ***@***.***>
Enviado:terça-feira, 4 de janeiro de 2022 12:17
Para: ***@***.***>
Cc:Rafael Gianini ***@***.***>; ***@***.***>
Assunto: Re: [balta-io/2801] Fundamentos do C# (Discussion #2)


@Rafael354-dev<https://github.com/Rafael354-dev> e qual seria o erro?

—
Reply to this email directly, view it on GitHub<https://github.com/balta-io/2801/discussions/2#discussioncomment-1906510>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ATYS5UJL2PRDN56UOV3VHPDUUMFO7ANCNFSM47UCVMGQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by Leonardocorreabrasil** [comment]
@andrebaltieri @Rafael354-dev 

Bom dia, debuguei  e realmente o valor do console ReadLine não converteu para o tipo double e não estou conseguindo converter.
![erro-2](https://user-images.githubusercontent.com/54727753/148203849-9c68c0c1-b2d2-48a6-90be-3e2353745a6b.PNG)

**Comment by marcelobarbieri** [comment]
Bom dia a todos. Leonardo, consegui conforme abaixo. Dá uma olhada.
[image: image.png]

Em qua., 5 de jan. de 2022 às 07:38, Leonardo Correa Brasil <
***@***.***> escreveu:

> @andrebaltieri <https://github.com/andrebaltieri> @Rafael354-dev
> <https://github.com/Rafael354-dev>
>
> Bom dia, debuguei e realmente o valor do console ReadLine não converteu
> para o tipo double e não estou conseguindo converter.
> [image: erro-2]
> <https://user-images.githubusercontent.com/54727753/148203849-9c68c0c1-b2d2-48a6-90be-3e2353745a6b.PNG>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/balta-io/2801/discussions/2#discussioncomment-1911222>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUTJXVQIKR2NBP6NF2TVBB3UUQNR7ANCNFSM47UCVMGQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Marcelo de Paiva Barbieri

**Reply by andrebaltieri** [reply]
@marcelobarbieri não veio imagem!

**Comment by marcelobarbieri** [comment]
Segue abaixo código e resultado obtido no terminal.

using System;

namespace Temp
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.Write("Número: ");
            double num = double.Parse(Console.ReadLine().Trim());
            Console.WriteLine(num.GetType());
            Console.WriteLine(num.GetTypeCode());
            Console.WriteLine(num);
            double result = num / 3.1;
            Console.WriteLine(result);
        }
    }
}




dotnet run
Número: 4,11
System.Double
Double
4,11
1,325806451612903
  ╱ Temp 


Em qua., 5 de jan. de 2022 às 08:52, Andre Baltieri <
***@***.***> escreveu:

> @marcelobarbieri <https://github.com/marcelobarbieri> não veio imagem!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/balta-io/2801/discussions/2#discussioncomment-1911535>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUTJXVQ3W6ZCME3OLKW53OTUUQWGZANCNFSM47UCVMGQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Marcelo de Paiva Barbieri

**Reply by andrebaltieri** [reply]
@marcelobarbieri obrigado pela resposta! Tem mais alguma ação a ser realizada nesta dúvida?

**Comment by Leonardocorreabrasil** [comment]
Bom dia @andrebaltieri @marcelobarbieri 

Obrigado pela força, debuguei e o problema pelo que eu percebi está na conversão do readLine, ![Capturar](https://user-images.githubusercontent.com/54727753/148541709-1b34212b-bd5f-4e2b-9d60-6e72cef4b502.PNG) fuçando ainda haha

**Reply by andrebaltieri** [reply]
Ai sim! Sensacional 💜

**Comment by Leonardocorreabrasil** [comment]
@andrebaltieri 

consegui amém, mas não sei se está correto o jeito que eu fiz.
![correto](https://user-images.githubusercontent.com/54727753/148545642-1ed08982-27bd-43ef-8b56-48c4bbf023ec.PNG)

**Reply by andrebaltieri** [reply]
Está sim 💜

**Comment by GeovaniDarcie** [comment]
Muito bom o curso! Muito obrigado.
Noite um possível erro quando o vídeo está em tela cheia o title não é inicializado pelo menos pra mim:
![image](https://user-images.githubusercontent.com/52720629/148591847-e9e7fa3c-3a79-4cc3-a01c-1767c1bc8ac0.png)

**Reply by andrebaltieri** [reply]
@GeovaniDarcie obrigado pelo feedback!
